java-frontend/src/main/java/org/sonar/java/model/SyntacticEquivalence.java

```diff
@@ -37,33 +37,53 @@ private SyntacticEquivalence() {
    * @return true, if nodes are syntactically equivalent
    */
   public static boolean areEquivalent(List<? extends Tree> leftList, List<? extends Tree> rightList) {
+    return areEquivalent(leftList, rightList, (t1, t2) -> false);
```

Comment:
I don't like the bellow variable name `equivalenceFilter`, I expect a filter to reduce the result and not to extends it. Here we pass `(t1, t2) -> false` to a filter, it's confusing, the result should be empty if it was a filter.
Why not calling it: `tolerantEquivalence` or `permissiveEquivalence`?